netboot/main.go

```diff
@@ -225,6 +241,40 @@ func boot(ifname string, dhcp dhcpFunc) error {
 	return nil
 }
 
+func getClientForBootfile(bootfile string) (*http.Client, error) {
+	var client *http.Client
+	if strings.HasPrefix(bootfile, "https://") {
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			debug("certs: rootCAs == nil")
+			rootCAs = x509.NewCertPool()
+		}
+		base64Certs, err := ioutil.ReadFile(*caCertFile)
+		decLen := base64.StdEncoding.DecodedLen(len(base64Certs))
+		certs := make([]byte, decLen)
+		n, err := base64.StdEncoding.Decode(certs, base64Certs)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to append %s to RootCAs: %v", *caCertFile, err)
+		}
+		if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+			debug("No VPD certs appended, using system certs only")
+		} else {
+			debug("%d bytes of VPD certs appended", n)
+		}
+		config := &tls.Config{
+			InsecureSkipVerify: *skipCertVerify,
+			RootCAs:            rootCAs,
+		}
+		tr := &http.Transport{TLSClientConfig: config}
+		client = &http.Client{Transport: tr}
+		debug("https client setup to use certs from VPD")
+	} else {
```

Comment:
the `else` should be defensively explicit to avoid errors when we will support more schemes

Reply:
getScheme now returns an error if we don't try http or https, so it'll return before this if. If someone has added a new scheme to getScheme that getClientForBootfile doesn't, it's now a fatal error.